lib/ansible/modules/cloud/azure/azure_rm_image.py

```diff
@@ -0,0 +1,291 @@
+    type: string
+status:
+    description: The result of the operation
+    returned: always
+    type: string
+'''
+
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.compute.models import Image
+    from azure.mgmt.compute.models import SubResource
+# from azure.mgmt.compute.models import VirtualMachineCaptureParameters
+except:
+    pass
+
+
+class AzureRMImage(AzureRMModuleBase):
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            vm_name=dict(
+                type='str',
+                required=False
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            location=dict(
+                type='str',
+                required=False
+            ),
+            state=dict(
+                type='str',
+                required=False,
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        required_if = [
+            ('state', 'present', ['vm_name'])
+        ]
+
+        self.resource_group = None
+        self.vm_name = None
+        self.name = None
+        self.state = None
+        self.location = None
+
+        self.results = dict(
+            changed=False,
+            ansible_facts=dict(azure_images=None)
+        )
+
+        super(AzureRMImage, self).__init__(
+            derived_arg_spec=self.module_arg_spec,
+            required_if=required_if,
+            supports_check_mode=True,
+            supports_tags=True)
+
+    def exec_module(self, **kwargs):
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        try:
+            resource_group = self.get_resource_group(self.resource_group)
+        except CloudError:
+            self.fail('resource group {} not found'.format(self.resource_group))
+        if not self.location:
+            self.location = resource_group.location
+        if self.state == 'present':
+            self.results = self.capture_image()
+        elif self.state == 'absent':
+            self.results = self.delete_image()
+        return self.results
+
+    def _list_images(self):
+        try:
+            images = self.compute_client.images
+            image_list = images.list()  # (resource_group_name=self.resource_group, name=self.name)
+        except CloudError:
+            self.fail("No images found!")
+        except Exception as e:
+            self.fail("An exception occurred: {}".format(str(e)))
+
+        named_images = []
+
+        for image in image_list:
+            image_info = dict(name=image.name,
+                              location=image.location,
+                              resource_group=image.id.split("/")[4],
```

Comment:
Is there an easier way to get the resource_group and not split the id?

Reply:
I hoped so too but we get back just the id.